src/app/modules/shared/components/breadcrumb/breadcrumb.component.spec.ts

```diff
@@ -1,27 +1,92 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { MatIconModule } from '@angular/material/icon';
+import { By } from '@angular/platform-browser';
+import { ActivatedRouteSnapshot } from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+import { MockService } from 'ng-mocks';
+import { Observable, of } from 'rxjs';
+import { EventsService } from 'src/app/modules/core/services/events.service';
+import { Breadcrumb, BreadcrumbService } from '../../services/breadcrumb.service';
 
 import { BreadcrumbComponent } from './breadcrumb.component';
 
 describe('BreadcrumbComponent', () => {
   let component: BreadcrumbComponent;
   let fixture: ComponentFixture<BreadcrumbComponent>;
+  let service: BreadcrumbService = MockService(BreadcrumbService);
+  const mockBreadcrumbs: Breadcrumb[] = [
+    {
+      displayName: 'Dashboard',
+      route: { path: 'dashboard' },
+      url: '/dashboard',
+    },
+    {
+      displayName: 'Wallet',
+      route: { path: 'wallet' },
+      url: '/dashboard/wallet',
+    },
+    {
+      displayName: 'Accounts',
+      route: { path: 'accounts' },
+      url: '/dashboard/wallet/accounts',
+    },
+    {
+      displayName: 'Create',
+      route: { path: 'create' },
+      url: '/dashboard/wallet/accounts/create',
+    },
+  ];
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      imports: [ MatIconModule ],
-      declarations: [ BreadcrumbComponent ]
+      imports: [
+        MatIconModule,
+        RouterTestingModule,
+      ],
+      declarations: [BreadcrumbComponent],
+      providers: [
+        EventsService,
+        { provide: BreadcrumbService, useValue: service },
+      ]
     })
     .compileComponents();
+    service = TestBed.inject(BreadcrumbService);
   }));
 
   beforeEach(() => {
+    service.create = (_: ActivatedRouteSnapshot): Observable<Breadcrumb[]> => {
+      return of(mockBreadcrumbs);
+    };
     fixture = TestBed.createComponent(BreadcrumbComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
   });
 
-  it('should create', () => {
+  it('should create the right number of breadcrumbs as list elements', () => {
```

Comment:
@wackerow please checkout these unit tests as a way to ensure our logic works as expected